### PR TITLE
Add pullapprove

### PR DESCRIPTION
**Pull request by robdodson**
- Removes CODEOWNERS
- Adds pullapprove config. Sets extensions team as owners for their docs with Simeon and Andy being default reviewers. All other docs fallback to the content team. Eng team owns all programmatic files.

### CODE REVIEW

**Comment by robdodson** [comment]
Before merging this we'll need to tweak some repo settings to turn off the required CODEOWNERS review and to set pullapprove as a required check.

File: .pullapprove.yml
Comment (samthor) [inline .pullapprove.yml:24]: It seems like these rules only exist to basically say _NOT_ the old nacl or mobile client docs. Is that right?

**Review by samthor — APPROVED** [review]

File: .pullapprove.yml
Comment (robdodson) [reply .pullapprove.yml:24]: Essentially. Though when we add devtools, workbox, etc, we'll probably need a similar block for them.

btw here's Angular's pullapprove file. They're like expert mode: https://github.com/angular/angular/blob/master/.pullapprove.yml

Probably some good ideas we can steal from them.

**Pull request merged**
